### PR TITLE
[Customer Portal][Web] Add Editable Update History Feature for Deployment Products with Full CRUD Support

### PR DESCRIPTION
**Pull request by dileepapeiris**
### Description

This pull request introduces a major enhancement to the deployment product management feature in the customer portal webapp. The primary change is the addition of an editable "Update History" tab for deployment products, allowing users to view, add, edit, and delete product updates. Supporting changes include API and model updates, UI adjustments, and comprehensive tests for the new functionality.

**Feature: Editable Update History for Deployment Products**
* Added `UpdateHistoryTab` component to `ManageProductModal`, enabling users to view, add, edit, and delete product updates for a deployment product. The tab is now enabled and fully functional, replacing the previous disabled state. (`apps/customer-portal/webapp/src/components/project-details/deployments/ManageProductModal.tsx` [[1]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL40-R41) [[2]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL64-R65) [[3]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL99-R100) [[4]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfR178-R200) [[5]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL219-R242) [[6]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfR293-R306) [[7]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL286-R320) [[8]](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL304-R338)
* Added a handler (`handleSaveUpdates`) to persist update history changes via the patch API. (`apps/customer-portal/webapp/src/components/project-details/deployments/ManageProductModal.tsx` [apps/customer-portal/webapp/src/components/project-details/deployments/ManageProductModal.tsxR178-R200](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfR178-R200))

**API and Model Updates**
* Extended the patch deployment product API and request model to support an `updates` field, allowing update history to be sent and stored. (`apps/customer-portal/webapp/src/api/usePatchDeploymentProduct.ts` [[1]](diffhunk://#diff-3604cf75a0a5c16b31944ee22d4e8ec06b9ced74e2e970a27b3fe30f086e0de1R62) `apps/customer-portal/webapp/src/models/requests.ts` [[2]](diffhunk://#diff-b6170f1b0e38899ced715e994dcc9ccc2325782694c87c87586271f6412933baR174-R178)
* Updated relevant types to include `ProductUpdate`. (`apps/customer-portal/webapp/src/components/project-details/deployments/ManageProductModal.tsx` [apps/customer-portal/webapp/src/components/project-details/deployments/ManageProductModal.tsxL40-R41](diffhunk://#diff-e46af29bb0bb64014102c4dcda986fac36b1e9c8d38277467c107327331793bfL40-R41))

**UI Adjustments**
* Removed the initial update information section from the add product modal to avoid redundancy with the new update history functionality. (`apps/customer-portal/webapp/src/components/project-details/deployments/AddProductModal.tsx` [apps/customer-portal/webapp/src/components/project-details/deployments/AddProductModal.tsxL483-L529](diffhunk://#diff-835c3f4386eff37dcd91eb2f9cb08ea6cde1e1ec7e5eb788f867da63fd325614L483-L529))

**Testing**
* Added a comprehensive test suite for the `UpdateHistoryTab` component, covering rendering, form interactions, loading states, and edge cases. (`apps/customer-portal/webapp/src/components/project-details/deployments/__tests__/UpdateHistoryTab.test.tsx` [apps/customer-portal/webapp/src/components/project-details/deployments/__tests__/UpdateHistoryTab.test.tsxR1-R110](diffhunk://#diff-e1f98a40371c09d9b29369a8fc5892f74bcfc11f6a5769596f58f22b4f419de6R1-R110))

**Documentation and Minor Improvements**
* Improved documentation and comments to clarify the new feature and its usage. (`apps/customer-portal/webapp/src/api/useGetRecommendedUpdateLevels.ts` [apps/customer-portal/webapp/src/api/useGetRecommendedUpdateLevels.tsL25-R27](diffhunk://#diff-a712d5378cbfa1fb871e4fb8e1ac6ed93ff04366ebb561fb802f80cdbad37113L25-R27))

These changes collectively provide a robust and user-friendly way to manage product update history within deployments.

## Summary by CodeRabbit

* **New Features**
  * Enabled an "Update History" tab with a timeline to add, edit, delete, and save product updates.
  * Persisted update-history changes from the product management modal.

* **Improvements**
  * Removed the initial update information section from the Add Product modal for a cleaner form.
  * Auto-fill descriptions from recommended update-level data when available.

* **Tests**
  * Added tests covering rendering, interactions, loading, and empty-state for the update history UI.

* **Documentation**
  * Updated inline API documentation text.

### CODE REVIEW

File: apps/customer-portal/webapp/src/components/project-details/deployments/UpdateHistoryTab.tsx [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+    [updateLevelsData],
+  );
+
+  const sortedUpdates = useMemo(() => {
+    return [...updates].sort((a, b) => {
+      const dateA = new Date(a.date).getTime();
+      const dateB = new Date(b.date).getTime();
+      return dateB - dateA;
+    });
+  }, [updates]);
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/components/project-details/deployments/UpdateHistoryTab.tsx]: Updated
